components/common/src/ui.rs

```diff
@@ -130,18 +132,18 @@ pub trait UIReader {
     fn prompt_yes_no(&mut self, question: &str, default: Option<bool>) -> Result<bool>;
 }
 
+pub trait ColorPrinter {
+    fn print(&mut self, buf: &[u8], color: Option<Color>, bold: bool) -> io::Result<()>;
```

Comment:
Do you mean `termcolor::NoColor` vs an Option<Color>?

Reply:
Yeah a weight enum seems like it would be more clear. I had thought about just passing a `ColorSpec` but it felt like that made alot of the calling code unnecesarily verbose.

Reply:
I think making the core interface totally general (i.e., `ColorSpec`) and building convenience functions into the trait on top of that to make our common usage ergonomic would be good. Most importantly, I don't want the callers to pass parameters values like `false` and `None`, when we can include more evocative text like `bold` and `yellow` (or, down the road a semantic identifier like `info` or `warning`)

Reply:
yeah we could add functions like `fn create_no_color_spec() -> ColorSpec` and `fn create_color_spec(color: Color, weight: Weight) -> ColorSpec`